File: src/service_client.py
import rospy
from tutor.srv import TanyaJawab, TanyaJawabResponse

if __name__ == "__main__":
    rospy.init_node("service_client")
    rospy.wait_for_service("tanya_jawab")
    try:
        service_client = rospy.ServiceProxy("tanya_jawab", TanyaJawab)
        tanya = input("Tanya: ")
        response = service_client(tanya)
        rospy.loginfo(f"Jawaban {response}")
    except rospy.ServiceException as e:
        rospy.logwarn(f"Service call failed: {e}")
    print("Done")

File: src/example/service_client.cpp
#include "ros/ros.h"
#include "tutor/TanyaJawab.h"


int main(int argc, char **argv)
{
  ros::init(argc, argv, "tanya_jawab_client");
  if (argc != 2)
  {
    ROS_INFO("usage: tanya aja!");
    return 1;
  }

  ros::NodeHandle n;
  ros::ServiceClient client = n.serviceClient<tutor::TanyaJawab>("tanya_jawab");
  tutor::TanyaJawab srv;
  srv.request.tanya = argv[1];
  if (client.call(srv))
  {
    ROS_INFO("Jawaban: %s", srv.response.jawab.c_str());
  }
  else
  {
    ROS_ERROR("Failed to call service tanya_jawab");
    return 1;
  }

  return 0;
}